GVFS/GVFS.Virtualization/Projection/GitIndexProjection.GitIndexEntry.cs

```diff
@@ -80,69 +97,136 @@ public unsafe void ParsePath()
+                            this.BuildingProjection_PathParts[partIndex] = LazyUTF8String.FromByteArray(pathPtr + currentPartStartIndex, i - currentPartStartIndex);
 
                             partIndex++;
                             currentPartStartIndex = i + 1;
 
-                            this.NumParts++;
-                            this.previousFinalSeparatorIndex = i;
+                            this.BuildingProjection_NumParts++;
+                            this.buildingProjectionPreviousFinalSeparatorIndex = i;
                         }
 
                         ++forLoopPtr;
                     }
 
                     // We unrolled the final part calculation to after the loop, to avoid having to do a 0-byte check inside the for loop
-                    this.PathParts[partIndex] = LazyUTF8String.FromByteArray(pathPtr + currentPartStartIndex, this.PathLength - currentPartStartIndex);
+                    this.BuildingProjection_PathParts[partIndex] = LazyUTF8String.FromByteArray(pathPtr + currentPartStartIndex, this.PathLength - currentPartStartIndex);
+
+                    this.BuildingProjection_NumParts++;
+                }
+            }
+
+            /// <summary>
+            /// Parses the path from the index as platform-specific relative path. 
+            /// It should only be called when running a background task.
+            /// </summary>
+            public unsafe void BackgroundTask_ParsePath()
+            {
+                this.PathBuffer[this.PathLength] = 0;
+
+                // The index in the buffer at which to start parsing
+                int loopStartIndex = 0;
+
+                // backgroundTaskRelativePathBuilder is reset to empty if the last index entry contained non-ASCII character(s)
+                // Only start from ReplaceIndex if the last entry contained *only* ASCII characters
+                if (this.backgroundTaskRelativePathBuilder.Length > 0)
+                {
+                    loopStartIndex = this.ReplaceIndex;
+                    this.backgroundTaskRelativePathBuilder.Length = this.ReplaceIndex;
+                }
+
+                fixed (byte* pathPtr = this.PathBuffer)
+                {
+                    byte* bufferPtrForLoop = pathPtr + loopStartIndex;
+                    while (loopStartIndex < this.PathLength)
+                    {
+                        if (*bufferPtrForLoop <= 127)
+                        {
+                            if (*bufferPtrForLoop == PathSeparatorCode)
+                            {
+                                this.backgroundTaskRelativePathBuilder.Append(GVFSPlatform.GVFSPlatformConstants.PathSeparator);
+                            }
+                            else
+                            {
+                                this.backgroundTaskRelativePathBuilder.Append(Convert.ToChar(*bufferPtrForLoop));

```

Comment:
`Convert.ToChar(byte)` is just doing a cast internally. Do you know if this call is getting inlined or not? It may be a worth a perf test with just a cast in here instead of this method call, to see if it makes a difference.